### PR TITLE
Compilable on gerty

### PR DESCRIPTION
**Pull request by jungan**
I don't know how to write `if or` logic in Makefile, so I just copy it. Welcome any suggestions on how to make it clean and concise.

### CODE REVIEW

File: src/mtcp/Makefile.in [hunk 1]
```diff
@@ -5,9 +5,6 @@ MANA_USE_LH_FIXED_ADDRESS = @MANA_USE_LH_FIXED_ADDRESS@
 CC = @CC@
 M32=@M32@
 CFLAGS = @CFLAGS@
-ifeq ($(findstring cori,$(PLATFORM)),cori)
-  CFLAGS += -DGNI
-endif
```
Comment (fyshhh) [inline src/mtcp/Makefile.in:10]: why is this removed?

File: contrib/mpi-proxy-split/Makefile_config.in [hunk 2]
```diff
@@ -20,6 +20,23 @@ default2: default
 ## But this .o file will be linked with src/dmtcp_coordinator.o, which
 ##   is built with g++.  So, we use $(CXX) for compatibility.
 ## An alternative is to use -O2, which inlines calls to memcpy.
+mana_coordinator.o: mana_coordinator.cpp
+	${CXX} ${CXXFLAGS} `CC --cray-print-opts=cflags` -c -o $@ $<
+else ifeq ($(findstring gert,$(PLATFORM)),gert)
```
Comment (fyshhh) [inline contrib/mpi-proxy-split/Makefile_config.in:25]: you could try setting a variable for each of cori and gerty, like `is_cori` and `is_gert` and then doing something like `ifeq ($(or $(is_cori),$(is_gert)), 1)` to reduce the amount of duplicated code.

File: src/mtcp/Makefile.in (hunk 1 above)
Comment (jungan) [reply src/mtcp/Makefile.in:10]: I cannot find anywhere using macro `GNI`, remove for simplicity.

File: contrib/mpi-proxy-split/Makefile_config.in (hunk 2 above)
Comment (jungan) [reply contrib/mpi-proxy-split/Makefile_config.in]: Good suggestion updated the code.

Comment (JainTwinkle) [reply contrib/mpi-proxy-split/Makefile_config.in]: @jungan should we make `is_cori` and `is_gerty` capital for consistency as all variables in the Makefile are in caps? Otherwise, LGTM. :)

Comment (jungan) [reply contrib/mpi-proxy-split/Makefile_config.in]: Sure, just updated.

**Review by JainTwinkle — APPROVED** [review]
Approving this one, but probably, we can move common and redundant variables from each Makefile to one common Makefile config in the future.
Thanks!

**Comment by jungan** [comment]
> Approving this one, but probably, we can move common and redundant variables from each Makefile to one common Makefile config in the future.
> Thanks!

Agreed, actually, I'm also thinking to do it in the `configure` so it will generate the correct Makefile on the current platform.
But I guess I need to focus on fixing bugs for now, merge it as it is, and will think about it later.
